finality-aleph/src/network/gossip/service.rs

```diff
@@ -40,19 +38,23 @@ enum Command<D: Data, P: Clone + Debug + Eq + Hash + Send + 'static> {
 ///   1. Messages are forwarded to the user.
 ///   2. Various forms of (dis)connecting, keeping track of all currently connected nodes.
 /// 3. Outgoing messages, sending them out, using 1.2. to broadcast.
-pub struct Service<N: RawNetwork, ES: EventStream<N::PeerId>, AD: Data, BSD: Data> {
-    network: N,
-    messages_from_authentication_user: mpsc::UnboundedReceiver<Command<AD, N::PeerId>>,
-    messages_from_block_sync_user: mpsc::UnboundedReceiver<Command<BSD, N::PeerId>>,
-    messages_for_authentication_user: mpsc::UnboundedSender<(AD, N::PeerId)>,
-    messages_for_block_sync_user: mpsc::UnboundedSender<(BSD, N::PeerId)>,
-    authentication_connected_peers: HashSet<N::PeerId>,
-    authentication_peer_senders: HashMap<N::PeerId, mpsc::Sender<AD>>,
-    block_sync_connected_peers: HashSet<N::PeerId>,
-    block_sync_peer_senders: HashMap<N::PeerId, mpsc::Sender<BSD>>,
+pub struct Service<
+    PeerId: Clone + Debug + Eq + Hash + Send + 'static,
```

Comment:
That's a bit inconvenient, consider creating an auto-trait? Unless you expect this will disappear completely after the proper abstraction is implemented, although I doubt it.

Reply:
let's leave it for now, I expect this struct to be heavily rewritten in the next PR